src/settings/MappingProfiles/MappingProfilesFormContainer/MappingProfilesFormContainer.js

```diff
@@ -15,6 +15,24 @@ import { MappingProfilesForm } from '../MappingProfilesForm';
 import { mappingProfileTransformations } from '../MappingProfilesTransformationsModal/TransformationsField/transformations';
 import { generateTransformationFieldsValues } from '../MappingProfilesTransformationsModal/TransformationsField';
 
+const isValidRecordTypesMatching = (selectedTransformations = [], selectedRecordTypes = []) => {
+  const intersection = {};
+
+  selectedTransformations.forEach(selectedTransformation => {
```

Comment:
I suggest simplifying the logic. It might not be as optimal as your solution, but way easier to reason about. It is enough to find only one transformation which does not match the selected record type.

```
selectedTransformations.find(({ recordType }) => !selectedRecordTypes.includes(recordType));
```

Based on my comment below additional logic might be added there to handle the `transformation value does need to be populated though`

```
selectedTransformations.find(({ recordType, transformation }) => transformation && !selectedRecordTypes.includes(recordType));
```

Reply:
This is a good idea. But we should check the difference between selected record types and used record types in transformations. So it means, that we should add extra check to handle the next cases:
1. There is a selected and filled transformation and missed selected record type
2. There is a selected record type and missed filled selected transformation.

Anyway, thanks for the good points.